### ISSUE TITLE
Lint errors cause molecule to skip cleanup

### ISSUE THREAD

**Issue by ghislainbourgeois** [ansible/molecule#1902]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible --version && molecule --version

ansible 2.7.9
  config file = None
  configured module search path = [u'/Users/ghibourg/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Sep 18 2018, 20:16:18) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
molecule, version 2.20.0
```

Molecule installation method (one of):

- pip

Ansible installation method (one of):

- pip

Detail any linters or test runners used:

- yaml-lint
- ansible-lint

- cleanup playbook defined

# Desired Behavior

Given `molecule test` with linters and a cleanup step configured, when the lint step errors out,
molecule should call the cleanup step and then the destroy step. 

# Actual Behaviour

Linting errors directly call `destroy`, never calling the `cleanup` step.

```
$ molecule init role -r test && cd test

$ cat molecule/default/molecule.yml
---
dependency:
  name: galaxy
driver:
  name: docker
lint:
  name: yamllint
platforms:
  - name: instance
    image: centos:7
provisioner:
  name: ansible
  lint:
    name: ansible-lint
  playbooks:
    cleanup: cleanup.yml
verifier:
  name: testinfra
  lint:
    name: flake8

$ cat molecule/default/cleanup.yml
---
- name: Cleanup
  hosts: localhost
  tasks:
    - name: Debug
      debug:
        msg: "Cleanup"

$ molecule test
--> Validating schema /Users/ghibourg/Projects/test/molecule/default/molecule.yml.
Validation completed successfully.
--> Test matrix
                                                                                                                                                                                                                                              └── default
    ├── lint
    ├── cleanup
    ├── destroy
    ├── dependency                                                                                                                                                                                                                                ├── syntax
    ├── create
    ├── prepare
    ├── converge
    ├── idempotence                                                                                                                                                                                                                               ├── side_effect
    ├── verify
    ├── cleanup
    └── destroy
                                                                                                                                                                                                                                              --> Scenario: 'default'
--> Action: 'lint'
--> Executing Yamllint on files found in /Users/ghibourg/Projects/test/...
Lint completed successfully.
--> Executing Flake8 on files found in /Users/ghibourg/Projects/test/molecule/default/tests/...
Lint completed successfully.
--> Executing Ansible Lint on /Users/ghibourg/Projects/test/molecule/default/playbook.yml...
    [701] Role info should contain platforms
    /Users/ghibourg/Projects/test/meta/main.yml:2
    {'meta/main.yml': {'__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'dependencies': [], u'galaxy_info': {u'description': u'your description', u'license': u'license (GPLv2, CC-BY, etc)', u'author': u'your name', u'company':
 u'your company (optional)', u'galaxy_tags': [], '__line__': 3, '__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'min_ansible_version': 1.2}, '__line__': 2}}

    [703] Should change default metadata: author
    /Users/ghibourg/Projects/test/meta/main.yml:2
    {'meta/main.yml': {'__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'dependencies': [], u'galaxy_info': {u'description': u'your description', u'license': u'license (GPLv2, CC-BY, etc)', u'author': u'your name', u'company':
 u'your company (optional)', u'galaxy_tags': [], '__line__': 3, '__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'min_ansible_version': 1.2}, '__line__': 2}}

    [703] Should change default metadata: description
    /Users/ghibourg/Projects/test/meta/main.yml:2
    {'meta/main.yml': {'__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'dependencies': [], u'galaxy_info': {u'description': u'your description', u'license': u'license (GPLv2, CC-BY, etc)', u'author': u'your name', u'company':
 u'your company (optional)', u'galaxy_tags': [], '__line__': 3, '__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'min_ansible_version': 1.2}, '__line__': 2}}
    [703] Should change default metadata: company
    /Users/ghibourg/Projects/test/meta/main.yml:2
    {'meta/main.yml': {'__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'dependencies': [], u'galaxy_info': {u'description': u'your description', u'license': u'license (GPLv2, CC-BY, etc)', u'author': u'your name', u'company':
 u'your company (optional)', u'galaxy_tags': [], '__line__': 3, '__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'min_ansible_version': 1.2}, '__line__': 2}}

    [703] Should change default metadata: license
    /Users/ghibourg/Projects/test/meta/main.yml:2
    {'meta/main.yml': {'__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'dependencies': [], u'galaxy_info': {u'description': u'your description', u'license': u'license (GPLv2, CC-BY, etc)', u'author': u'your name', u'company':
 u'your company (optional)', u'galaxy_tags': [], '__line__': 3, '__file__': u'/Users/ghibourg/Projects/test/meta/main.yml', u'min_ansible_version': 1.2}, '__line__': 2}}

An error occurred during the test sequence action: 'lint'. Cleaning up.
--> Scenario: 'default'
--> Action: 'destroy'

    PLAY [Destroy] *****************************************************************

    TASK [Destroy molecule instance(s)] ********************************************
    changed: [localhost] => (item=None)
    changed: [localhost]

    TASK [Wait for instance(s) deletion to complete] *******************************
    ok: [localhost] => (item=None)
    ok: [localhost]

    TASK [Delete docker network(s)] ************************************************

    PLAY RECAP *********************************************************************
    localhost                  : ok=2    changed=1    unreachable=0    failed=0

```

**Comment by ghislainbourgeois** [comment]
I am currently working on a PR for this issue.

**Comment by decentral1se** [comment]
Good catch! /cc @ragingpastry for review/comments etc. from https://github.com/ansible/molecule/pull/1676

**Comment by seandst** [comment]
The changes in #1739 would likely conflict with changes made for this issue, and #1739 is also affected by this issue. If desired, I would be happy to add a change to that pull request to remember to run cleanup before destroy on error, and put a couple of tests in to cover the new behavior.

**Comment by seandst** [comment]
I totally missed that #1903 was opened, so I'll be integrating this into #1739 after merge.

**Comment by decentral1se** [comment]
OK, thanks @seandst!

**Comment by decentral1se** [comment]
OK, we merged #1903, let's close this off.
